webrtc.html

```diff
@@ -7374,6 +7443,72 @@ <h2>Methods</h2>
               A". However, <span data-jsep="initial-answers">[[!JSEP]]</span>
               allows adding codecs that were not in the offer, so
               implementations can behave differently.</p>
+              <p>When <code>setCodecPreferences()</code> in invoked, the <a>user agent</a>
+              MUST run the following steps:</p>
+              <ol>
+                <li>
+                  <p>Let <var>transceiver</var> be the <code><a>RTCRtpTransceiver</a></code>
+                  object this method was invoked on.</p>
+                </li>
+                <li>
+                  <p>Let <var>codecs</var> be the first argument.</p>
+                </li>
+                <li>
+                  <p>If <var>codecs</var> is an empty list, set set <var>transceiver</var>'s
+                  <a>[[\PreferredCodecs]]</a> slot to <var>codecs</var> and abort these steps.</p>
+                </li>
+                <li>
+                  <p>Remove any duplicate values in <var>codecs</var>. Start at the back of the
+                  list such that the priority of the codecs is maintained; the index of the
+                  first occurrence of a codec within the list is the same before and after this
+                  step.</p>
+                </li>
+                <li>
+                  <p>Let <var>kind</var> be the <var>transceiver</var>'s
+                  <a>transceiver kind</a>.</p>
+                </li>
+                <li>
+                  <p>If the intersection between <var>codecs</var> and
+                  <code>RTCRtpSender.getCapabilities(kind).codecs</code> or the intersection
+                  between <var>codecs</var> and
+                  <code>RTCRtpReceiver.getCapabilities(kind).codecs</code> is an empty set,
+                  throw <code>InvalidModificationError</code>. This ensures that we always have
+                  something to offer, regardless of <code><a>transceiver</a><a>direction</a></code>.</p>
```

Comment:
If we don't do this then we would have to consider throwing an exception when you set "direction" or when doing createOffer() or createAnswer(); or we would have to have a fallback-plan at createOffer/Answer to say that if filtering based on direction results in an empty list of preferences, act as if no preferences was set.